tests/lib/rules/no-missing-import.js

```diff
@@ -87,6 +87,18 @@ ruleTester.run("no-missing-import", rule, {
             filename: fixture("test.js"),
             code: "import mocha from 'mocha';",
         },
+        {
+            filename: fixture("test.js"),
+            code: "import something from 'cjs-module-with-no-main';",
+        },
+        {
+            filename: fixture("test.js"),
+            code: "import something from 'esm-module';",
+        },
+        {
+            filename: fixture("test.js"),
+            code: "import something from 'esm-module/sub';",
+        },
```

Comment:
Can you add some invalid cases?

Reply:
Will do.

Reply:
Oh, shit. The failing test opened a hornet's nest of problems. Will get back to you in a few... :-)

Reply:
OK, that "invalid case" test failed because it was valid. The code allows things that Node.js ESM just doesn't allow. So I changed it, and now other tests that passed do NOT pass, because they allowed things that are not allowed! So this is turning into quite the work for me. It'll take me a few days. Will definitely get back to you on this one.

Reply:
Sorry, I should have  be clearer. I mean not using the right esm export. e.g.
```json
{"exports": {".": "src/main.js"}}
```
```js
import mod from "esm-module/src/main"; // => error
import mod from "esm-module/src/main.js"; // => error
```